### PR TITLE
Use MuxerPath to determine location to dotnet.exe

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets [hunk 1]
```diff
@@ -51,7 +58,7 @@
   <Target
     Name="_AddDesktopReferences"
     AfterTargets="ResolveLockFileReferences"
-    Condition="'$(TargetFrameworkIdentifier)'=='.NETFramework'">
+    Condition="'$(_MvcViewCompilationAddDesktopReferences)'!='false' AND '$(TargetFrameworkIdentifier)'=='.NETFramework'">
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets]: This target gets called too early and results in `_MvcViewCompilationBinaryPath` being not set by common-testapps. Results in warnings when running tests. We'll use MusicStore to get coverage for this.

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tasks/GetDotNetHost.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tasks/GetDotNetHost.cs]: Shouldn't need all these usings.

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tasks/GetDotNetHost.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+namespace Microsoft.AspNetCore.BuildTools
+{
+    public class GetDotNetHost : Task
+    {
+        [Output]
+        public string MuxerPath { get; set; }
+
+        public override bool Execute()
+        {
+            MuxerPath = DotNetMuxer.MuxerPathOrDefault();
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tasks/GetDotNetHost.cs]: Loosely based on https://github.com/aspnet/BuildTools/blob/dev/src/Internal.AspNetCore.BuildTools.Tasks/GetDotNetHost.cs. We'll fallback to using the exe from the path in the event the user is using desktop msbuild to publish a netcoreapp2.0 application.

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tasks/GetDotNetHost.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+using System;
+using System.Runtime.InteropServices;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Extensions.CommandLineUtils;
+
+namespace Microsoft.AspNetCore.BuildTools
+{
+    public class GetDotNetHost : Task
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tasks/GetDotNetHost.cs]: I'm shocked there isn't an already present MSBuid property containing this value.

File: build/common-testapps.props [hunk 5]
```diff
@@ -1,7 +1,9 @@
 <Project>
   <Import Project="dependencies.props" />
   <PropertyGroup>
+    <_MvcViewCompilationAddDesktopReferences>false</_MvcViewCompilationAddDesktopReferences>
```
Comment (NTaylorMullen) [inline build/common-testapps.props]: Why we need this guy?

Comment (pranavkm) [inline build/common-testapps.props]: See https://github.com/aspnet/MvcPrecompilation/pull/150#discussion_r122851384

**Comment by pranavkm** [comment]
🆙 📅 and added new project to sln.

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets (hunk 1 above)
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets]: We should figure out a way to go back to testing the tools via packages. There's so much hack in general to make the project reference approach work.

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tasks/GetDotNetHost.cs (hunk 3 above)
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tasks/GetDotNetHost.cs]: Does the MSBuild infrastructure do this out of the box for us and shove it into a property?

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tasks/GetDotNetHost.cs]: Nope - https://github.com/Microsoft/msbuild/issues/1669

Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tasks/GetDotNetHost.cs]: Wow total bummer.

**Pull request closed without merging**